### PR TITLE
feature(product) attribute relation add

### PR DESCRIPTION
**Pull request by rprzedzik**
# Description

Add new attribute type: product relation

Issue #1384
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Unit Tests
- [x] Behat Test

# Checklist:
- [x] I have read the contribution requirements and fulfill them.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: module/product/src/Infrastructure/Factory/Command/Update/UpdateProductRelationAttributeCommandFactory.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+class UpdateProductRelationAttributeCommandFactory implements UpdateAttributeCommandFactoryInterface
+{
+    public function support(string $type): bool
+    {
+        return $type === ProductRelationAttribute::TYPE;
+    }
+
+    /**
+     * @param FormInterface $form \
```
Comment (piotrkreft) [inline module/product/src/Infrastructure/Factory/Command/Update/UpdateProductRelationAttributeCommandFactory.php:31]:
```suggestion
     * @param FormInterface $form
```

File: module/product/src/Resources/translations/attribute.en.yaml [hunk 2]
```diff
@@ -0,0 +1 @@
+PRODUCT-RELATION: Relation
```
Comment (piotrkreft) [inline module/product/src/Resources/translations/attribute.en.yaml:1]: Product relation? Do not know whether it's worth being that exact

**Review by piotrkreft — CHANGES_REQUESTED** [review]

File: module/product/src/Resources/translations/attribute.en.yaml (hunk 2 above)
Comment (rprzedzik) [reply module/product/src/Resources/translations/attribute.en.yaml:1]: IMHO no.

**Review by piotrkreft — DISMISSED** [review]

File: module/product/tests/Infrastructure/Factory/Command/Create/CreateProductRelationAttributeCommandFactoryTest.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    public function testCreation(): void
+    {
+        $data = $this->getAttributeFormModel(AttributeFormModel::class);
+        $form = $this->createMock(FormInterface::class);
+        $form->method('getData')->willReturn($data);
+
+        $commandFactory = new CreateProductRelationAttributeCommandFactory();
+
+        /** @var CreateGalleryAttributeCommand $result */
```
Comment (wfajczyk) [inline module/product/tests/Infrastructure/Factory/Command/Create/CreateProductRelationAttributeCommandFactoryTest.php:35]:
```suggestion
        /** @var CreateProductRelationAttributeCommand $result */
```

File: module/product/tests/Infrastructure/Factory/Command/Update/UpdateProductRelationAttributeCommandFactoryTest.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    public function testCreation(): void
+    {
+        $id = $this->createMock(AttributeId::class);
+        $data = $this->getAttributeFormModel(AttributeFormModel::class);
+        $form = $this->createMock(FormInterface::class);
+        $form->method('getData')->willReturn($data);
+
+        $commandFactory = new UpdateProductRelationAttributeCommandFactory();
+
+        /** @var UpdateGalleryAttributeCommand $result */
```
Comment (wfajczyk) [inline module/product/tests/Infrastructure/Factory/Command/Update/UpdateProductRelationAttributeCommandFactoryTest.php:37]:
```suggestion
        /** @var UpdateProductRelationAttributeCommand $result */
```

**Review by wfajczyk — CHANGES_REQUESTED** [review]

**Review by wiewiurdp — APPROVED** [review]

**Review by wfajczyk — APPROVED** [review]

**Pull request merged**